DOCS-9614: This is the match Statement, it's not a function (#1717)
mulesoft/docs-mule-runtime: modules/ROOT/pages/dataweave-pattern-matching.adoc

}
----

> For use cases in which the input is a String, you can also use the xref:dw-core-functions-match.adoc[match] _function_ in the `dw::Core` module to test the string against a regex pattern. 

Reviewer: ```suggestion
For use cases in which the input is a String, you can also use the xref:dw-core-functions-match.adoc[match] _function_ in the `dw::Core` module to test the string against a regex pattern (it returns an array with the string that matches the entire regex and its capture groups). 
```
Something like that maybe. (a little detail)
I think that the difference between match and matches functions wasn't clear.

Reviewer 2: Converted the parenthetical into a regular sentence.

> For use cases in which the input is a String, you can also use the xref:dw-core-functions-match.adoc[match] _function_ in the `dw::Core` module to test the string against a regex pattern. 

Reviewer 2: ```suggestion
For use cases in which the input is a String, you can also use the xref:dw-core-functions-match.adoc[match] _function_ in the `dw::Core` module to test the string against a regex pattern. The function returns an array with the string that matches the entire regex and its capture groups.
```

> For use cases in which the input is a String, you can also use the  xref:dw-core-functions-match.adoc[match] _function_ in the `dw::Core` module to test the string against a regex pattern. 

Reviewer 2: ```suggestion
For use cases in which the input is a String, you can also use the xref:dw-core-functions-match.adoc[match] _function_ in the `dw::Core` module to test the string against a regex pattern. 
```

> For use cases where you need a boolean result based on whether a String left-side input argument matches some regex pattern, you can use the `matches` function instead of using pattern matching or the dw::Core module's `match` function.

Reviewer 2: ```suggestion
For use cases that require a Boolean result based on whether an input argument matches a regex pattern, you can use the  xref:dw-core-functions-matches.adoc[matches] _function_ in the `dw::Core` module instead of using pattern matching or the `match` function.
```

> For use cases that require a Boolean result based on whether a String  matches a regex pattern, you can use the  xref:dw-core-functions-matches.adoc[matches] _function_ in the `dw::Core` module instead of using pattern matching or the dw::Core module's `match` function.

Reviewer 2: ```suggestion
For use cases that require a Boolean result based on whether a String  matches a regex pattern, you can use the xref:dw-core-functions-matches.adoc[matches] _function_ in the `dw::Core` module instead of using pattern matching or the dw::Core module's `match` function.
```

}
----

> ==Other Ways to Pattern Match a String

Reviewer 2: ```suggestion

```

> For use cases where the left-side input argument is a String, you can also use the dw::Core module's `match` function to test the string against a regex pattern. 

Reviewer 2: ```suggestion
For use cases in which the input is a String, you can also use the  xref:dw-core-functions-match.adoc[match] _function_ in the `dw::Core` module to test the string against a regex pattern. 
```

> For use cases where you need a boolean result based on whether a String left-side input argument matches some regex pattern, you can use the `matches` function instead of using pattern matching or the dw::Core module's `match` function.

Reviewer 2: ```suggestion
For use cases that require a Boolean result based on whether a String  matches a regex pattern, you can use the  xref:dw-core-functions-matches.adoc[matches] _function_ in the `dw::Core` module instead of using pattern matching or the dw::Core module's `match` function.
```
